src/docbuild/utils/pidlock.py

```diff
@@ -1,169 +1,151 @@
 """PID file locking utility."""
 
-import atexit
-import hashlib
-import logging
-import os
 import errno
+import fcntl
+import io
+import logging
 import os
 from pathlib import Path
-from typing import Self, Any
+from typing import ClassVar, Dict, Self, cast
```

Comment:
You don't need `Dict` as we are using Python 3.12+


```suggestion
from typing import ClassVar, Self, cast
```